Tests/BasicTests/FileSystemTests.swift

```diff
@@ -42,10 +43,23 @@ class FileSystemTests: XCTestCase {
         XCTAssertTrue(fs.isFile(sym))
         XCTAssertFalse(fs.isDirectory(sym))
 
+        // isExecutableFile
+        let executable = tempDir.path.appending(component: "exec-foo")
+        let stream = BufferedOutputByteStream()
+        stream <<< "#!/bin/sh" <<< "\n"
+        stream <<< "set -e" <<< "\n"
+        stream <<< "exit" <<< "\n"
+        try! localFileSystem.writeFileContents(executable, bytes: stream.bytes)
+        _ = try! Process.popen(args: "chmod", "+x", executable.asString)
```

Comment:
use `Process.checkNonZeroExit()` because this will not fail if chmod exits with non zero code.